typescript/src/client/solana/solana.ts

```diff
@@ -72,6 +78,72 @@ export class SolanaClient implements SolanaClientInterface {
     return account;
   }
 
+  /**
+   * Exports a CDP Solana account's private key.
+   * It is important to store the private key in a secure place after it's exported.
+   *
+   * @param {ExportAccountOptions} options - Parameters for exporting the Solana account.
+   * @param {string} options.address - The address of the account.
+   * @param {string} options.name - The name of the account.
+   *
+   * @returns A promise that resolves to the exported account's private key
+   * which is a base58 encoding of the account's full 64-byte private key.
+   *
+   * @example **With an address**
+   * ```ts
+   * const privateKey = await cdp.solana.exportAccount({
+   *   address: "1234567890123456789012345678901234567890",
+   * });
+   * ```
+   *
+   * @example **With a name**
+   * ```ts
+   * const privateKey = await cdp.solana.exportAccount({
+   *   name: "MyAccount",
+   * });
+   * ```
+   */
+  async exportAccount(options: ExportAccountOptions): Promise<string> {
+    if (!options.address && !options.name) {
+      throw new Error("Either address or name must be provided");
+    }
+
+    try {
+      const { publicKey, privateKey } = await generateExportEncryptionKeyPair();
+
+      if (options.address) {
+        const { encryptedPrivateKey } = await CdpOpenApiClient.exportSolanaAccount(
+          options.address,
+          {
+            exportEncryptionKey: publicKey,
+          },
+          options.idempotencyKey,
+        );
+        const decryptedPrivateKey = decryptWithPrivateKey(privateKey, encryptedPrivateKey);
+        return formatSolanaPrivateKey(decryptedPrivateKey);
+      }
+
+      if (options.name) {
+        const { encryptedPrivateKey } = await CdpOpenApiClient.exportSolanaAccountByName(
+          options.name,
+          {
+            exportEncryptionKey: publicKey,
+          },
+          options.idempotencyKey,
+        );
+        const decryptedPrivateKey = decryptWithPrivateKey(privateKey, encryptedPrivateKey);
+        return formatSolanaPrivateKey(decryptedPrivateKey);
+      }
+
+      throw new Error("Either address or name must be provided");
```

Comment:
This line can be removed, since you have the validation at the top of the function